### DISCUSSION TITLE
Orientation of a matrix

### DISCUSSION THREAD

**Discussion by RFLTools** [PlummersSoftwareLLC/NightDriverStrip#236] [Q&A] [answered]
Is there a way to simply change the start position of a pixel?  i.e. revise the Pixel 0 position from TOP-LEFT to, say, LOWER-RIGHT?

I ask because I'm reusing a WLED project where I constructed a 3D printed frame using 3 16x16 panels and the spectrum project in NightDriver is upside-down.  I'd prefer not to have to rip apart my frame to reinstall the panels rotated 180 degrees.

**Comment by rbergen** [comment]
If I read your question correctly, you're actually looking for a way to rotate the orientation of the matrix by 180 degrees. I didn't test this, but you could try adding the following line to the very start of the [`LEDMatrixGFX::StartMatrix()` function in ledmatrixgfx.cpp](https://github.com/PlummersSoftwareLLC/NightDriverStrip/blob/8ab63637eda5712568dc6eca87c7ea128f2f7569/src/ledmatrixgfx.cpp#L57):

```cpp
    matrix.setRotation(rotation180);
```

**Comment by RFLTools** [comment, marked as answer]
You read correctly but unfortunately the Spectrum doesn't use he Matrix libraries.  As such I did find a solution by adding:
`x = _width - x - 1; y = _height - y - 1;`
to the start of
`inline virtual uint16_t xy(uint16_t x, uint16_t y) const`
Which would essentially accomplish the same as rotating 180 degrees.

Oh - and thanks for the prompt reply!

Edit:  Just to be clear if anyone is trying to resolve the same issue, the changes were made to gfxbase.h

**Reply by rbergen** [reply]
Well, based on how useful my reply was, next time I should probably spend more time looking into things and answer more slowly. Anyway, I'm glad you found a solution that _does_ work!

**Comment by davepl** [comment]
Good point - you can do whatever remapping (rotations, flips, mirrors) you like in the xy virtual override.

**Comment by RFLTools** [comment]
Oh, and just a comment off topic:  Those M5StickC Plus controllers are fantastic!

I was going to use a more standard ESP32 with an I2S mic but your videos made me aware of them - great little devices with so many features (and quite inexpensive considering all the features).

**Comment by davepl** [comment]
Check out their website, they have a TON of cool modules and adding!

https://m5stack.com/￼

**Comment by RFLTools** [comment]
lol - just what I need to look into after my wife's instruction to "stop buying $hit"...

**Comment by davepl** [comment]
OT: I once saw an eBay auction that offered to include a “cheap” invoice “for your wife”.  No idea if he was kidding...
